datacube/ui/task_app.py

```diff
@@ -263,6 +263,14 @@ def add_dataset_to_db(index, datasets):
 def do_nothing(result):
     pass
 
+def _wrap_impl(f, args, kwargs, task):
+    'needs to be at the top level'
+    return f(task, *args, **kwargs)
+
+def wrap_task(f, *args, **kwargs):
+    'turn function `f(task, *args, **kwargs)` into `g(task)` in pickle-able fashion'
+    return functools.partial(_wrap_impl, f, args, kwargs)
+
```

Comment:
We should aim for PEP 257 with docstrings - Triple quotes are used even though the string fits on one line. This makes it easy to later expand it.